cli/module_generate/modulegen/inputs.go

```diff
@@ -85,15 +87,40 @@ func (inputs *ModuleInputs) HasEmptyInput() bool {
 	return false
 }
 
-// CheckResource checks if the given resource is valid.
-func (inputs *ModuleInputs) CheckResource() error {
-	if inputs.ResourceSubtype == "" || inputs.ResourceType == "" {
+// SpaceReplacer removes spaces, dashes, and underscores from a string
+var SpaceReplacer = strings.NewReplacer(" ", "", "_", "", "-", "")
+
+// CheckResourceAndSetType checks if the given resource subtype is valid, and sets the corresponding resource type if so.
+func (inputs *ModuleInputs) CheckResourceAndSetType() error {
+	if inputs.ResourceSubtype == "" {
 		return nil
 	}
+	if inputs.ResourceSubtype == "generic" {
+		return fmt.Errorf(
+			"resource subtype 'generic' cannot be differentiated; please specify either 'generic-service' or 'generic-component'")
+	}
 	for _, resource := range Resources {
-		if inputs.Resource == resource {
+		splitResource := strings.Split(resource, " ")
+		if len(splitResource) != 2 {
+			return errors.New("resource not formatted correctly in code base; this shouldn't happen. please consider filing a ticket")
+		}
+		subtype := splitResource[0]
+		// make sure we support subtypes that are passed with different spacers (e.g.,
+		// "power sensor", "power-sensor", "power_sensor", "powerSensor")
```

Comment:
a bit of flyby logic here. previously, the CLI would complain if we passed `power-sensor` but not `power_sensor`. Changes here enable us to be more case-agnostic. for multi-word resource subtypes.